great_expectations/core/http_handler.py

```diff
@@ -0,0 +1,77 @@
+from typing import Callable, Optional
+
+import requests
+
+from great_expectations import __version__
+
+
+class HTTPResponse:
+    def __init__(self, response: requests.Response) -> None:
+        self._response = response
+
+    @property
+    def status_code(self) -> int:
+        return self._response.status_code
+
+    @property
+    def text(self) -> str:
+        return self._response.text
+
+    def json(self) -> dict:
+        return self._response.json()
+
+    def raise_for_status(self) -> None:
+        self._response.raise_for_status()
+
+
+class HTTPHandler:
+    """
+    Facade class designed to be a lightweight wrapper around HTTP requests.
+    For all external HTTP requests made in Great Expectations, this is the entry point.
+
+    HTTPHandler is designed to be library agnostic - the underlying implementation does not
+    matter as long as we fulfill the following contract:
+        * get
+        * post
+        * put
+        * patch
+        * delete
+    """
+
+    TIMEOUT = 20
+
+    def __init__(self, access_token: str, timeout: int = TIMEOUT) -> None:
+        self._headers = self._init_headers(access_token)
+        self._timeout = timeout
+
+    def _init_headers(self, access_token: str) -> dict:
+        headers = {
+            "Content-Type": "application/vnd.api+json",
+            "Authorization": f"Bearer {access_token}",
+            "Gx-Version": __version__,
+        }
+        return headers
+
+    def _make_request(
+        self,
+        fn: Callable,
```

Comment:
Can you annotate what this callable's inputs and outputs? That is `Callable[[input type list], output type]`.